### PR TITLE
rework band info creation to make sure output contains all selector values

### PR DESCRIPTION
**Pull request by Shane98c**
This fixes the issue we were seeing when trying to select down a store containing multiple bands, but also other selectors that need to be passed through. This should now include all the selectors with each band (eg elapsed_time, ids, etc)

I've tested this with the seaweed viewer and it seems to behave correctly. It's also working with OAE.

### CODE REVIEW

**Review by katamartin — COMMENTED** [review]
Thanks for tackling this! I have to confess I find the logic sort of hard to follow (probably due to the way I set this up 🙈), so I quickly wrote up some tests to compare what's happening against `main`.

Want to merge these in? https://github.com/carbonplan/maps/pull/109

File: src/utils.js [hunk 1]
```diff
@@ -365,36 +365,35 @@ export const getPyramidMetadata = (multiscales) => {
...
-              [selectorKey]: values[i],
-            }
-          })
-        } else {
-          updatedBands[key] = { [selectorKey]: values[i] }
-        }
+    })
+    combinedBands = newCombinedBands
+  })
+  Object.keys(selector).forEach((key) => {
```
[the remark this answers is not in the document]
Comment (Shane98c) [reply src/utils.js:398]: no functional changes with the rewrite above so reverted to keep this update a little cleaner! i wasn't parsing the reducer well last night

[the remark this answers is not in the document]
Comment (katamartin) [reply src/utils.js:398]: ooh gotcha. if the reducer was feeling more readable to you with the changes, totally open to leaving em in!

**Review by katamartin — APPROVED** [review]
🚢
